pkg/logs/input/container/docker.go

```diff
@@ -226,7 +227,9 @@ func (dt *DockerTailer) checkForNewDockerTags() {
 		log.Warn(err)
 	} else {
 		if !reflect.DeepEqual(tags, dt.containerTags) {
-			dt.containerTags = tags
+			// Copy the slice to avoid side effects when append-ing the other tags
+			dt.containerTags = make([]string, len(tags))
+			copy(dt.containerTags, tags)
```

Comment:
not sure why we're copying there, is it just a safety measure? we don't do it in the core check

Reply:
As the slice is passed down and `append`ed later, see https://github.com/DataDog/datadog-agent/blob/master/pkg/logs/message/origin.go#L29-L40 , I went with the safest route, as:

- it is less intrusive than patching the rest of the downstream processing
- we are behind a caching later (`if !reflect.DeepEqual(tags, dt.containerTags)`)